server/server.go

```diff
@@ -16,38 +16,17 @@
 // AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 // LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
-// THE SOFTWARE.
+// THE SOFTWARE
 
 package server
 
-import (
-	"net/http"
+// Server is a server capable of listening to incoming traffic and closing itself
+// when it's shut down.
+type Server interface {
```

Comment:
Why make this into an interface instead of leaving as a struct? I don't have much context, but it's seems like there would only ever be one server, no? This is the [approach taken by the Go standard library](https://golang.org/pkg/net/http/#Server) as well.

Reply:
I could've, but it'll make testing more difficult because you can't mock it as a struct. Go standard library worked around this by providing `httptest`, but that's a completely different approach from what we usually do for unit testing.

Was also thinking maybe at some point there could be a GRPC server for communicating with internal services.